### PR TITLE
Enable the building preview image in construction placement.

### PR DESCRIPTION
**Pull request by adamjryan**

### CODE REVIEW

**Comment by adamjryan** [comment]
Looks like it was commented out previously (although it needed some updating to get working again), any reason?

File: jsettlers.graphics/src/jsettlers/graphics/map/MapContent.java [hunk 1]
```diff
@@ -410,23 +417,13 @@ private void drawMain(FloatRectangle screen) {
...
-		// map.getConstructionPreviewBuilding().getImages()) {
-		// ImageProvider.getInstance().getImage(image).draw(context.getGl());
-		// }
-		// context.endTileContext();
-		// }
-		// }
+        if (placementBuilding != null) {
+            ShortPoint2D underMouse = this.context.getPositionOnScreen((float) mousePosition.getX(), (float) mousePosition.getY());
+            IMapObject mapObject = context.getMap().getMapObjectsAt(underMouse.x, underMouse.y);
+            if (mapObject != null && mapObject.getObjectType() == EMapObjectType.CONSTRUCTION_MARK) {
```
Comment (michaelzangl) [inline jsettlers.graphics/src/jsettlers/graphics/map/MapContent.java]: Mind that map objects are stored in a linked list. This map object could have a child that is a Construction mark. Use a while-loop for this instead and loop through all children.

**Comment by michaelzangl** [comment]
Can you have a look at the "controls-improve-menu"-branch? I started to change a lot of the building marks state handling but changes should be easy to port. The main change is that the build panel on the left keeps track of the building to build.

**Comment by adamjryan** [comment]
Created a new pull request for the controls-improve-menu branch, cheers.

**Pull request closed without merging**